server/src/main/java/org/opensearch/extensions/ExtensionStringResponse.java

```diff
@@ -15,18 +15,18 @@
 import java.io.IOException;
 
 /**
- * Response to register REST Actions request.
+ * Generic string response indicating the status of some previous request sent to the SDK
  *
  * @opensearch.internal
  */
-public class RegisterRestActionsResponse extends TransportResponse {
+public class ExtensionStringResponse extends TransportResponse {
```

Comment:
Hm .. what this class is for? What purpose the "generic string response" bears?

Reply:
During our initialization there are a few times when we just need a simple response such as a boolean (success/failure) or String (value of a configuration setting).   We use the `ExtensionStringResponse` for:
 - Registering REST handlers on the `RestController` (we get a response listing the handlers which are registered, which is then logged)
 - Registering custom settings (we get a response listing the settings that were registered, which is then logged)

Sample log messages resulting from this String Response:
> 09:13:31.552 [opensearch[ad-extension][generic][T#2]] INFO  org.opensearch.sdk.handlers.ExtensionStringResponseHandler - received Registered extension ad to handle REST Actions [POST /detectors, GET /detectors, POST /detectors/_validate, POST /detectors/_validate/{type}]
> 09:13:31.559 [opensearch[ad-extension][generic][T#4]] INFO  org.opensearch.sdk.handlers.ExtensionStringResponseHandler - received Registered settings from extension ad: opendistro.anomaly_detection.enabled, opendistro.anomaly_detection.breaker.enabled, plugins.anomaly_detection.enabled, plugins.anomaly_detection.breaker.enabled, plugins.anomaly_detection.hcad_cold_start_interpolation.enabled, plugins.anomaly_detection.max_entities_for_preview, plugins.anomaly_detection.page_size, plugins.anomaly_detection.ad_result_history_max_docs_per_shard, plugins.anomaly_detection.ad_result_history_rollover_period, plugins.anomaly_detection.ad_result_history_retention_period, plugins.anomaly_detection.max_primary_shards, plugins.anomaly_detection.model_max_size_percent, plugins.anomaly_detection.max_retry_for_unresponsive_node, plugins.anomaly_detection.backoff_minutes, plugins.anomaly_detection.checkpoint_write_queue_max_heap_percent, plugins.anomaly_detection.checkpoint_write_queue_concurrency, plugins.anomaly_detection.checkpoint_write_queue_batch_size, plugins.anomaly_detection.cooldown_minutes, plugins.anomaly_detection.max_old_ad_task_docs_per_detector, plugins.anomaly_detection.batch_task_piece_interval_seconds, plugins.anomaly_detection.delete_anomaly_result_when_delete_detector, plugins.anomaly_detection.max_batch_task_per_node, plugins.anomaly_detection.max_running_entities_per_detector_for_historical_analysis, plugins.anomaly_detection.request_timeout, plugins.anomaly_detection.filter_by_backend_roles, plugins.anomaly_detection.detection_interval, plugins.anomaly_detection.detection_window_delay, plugins.anomaly_detection.max_anomaly_detectors, plugins.anomaly_detection.max_multi_entity_anomaly_detectors, plugins.anomaly_detection.max_anomaly_features

Reply:
It sounds like "proxy for logging". I don't think we should be doing that: each extension has own its process + log, where such information has to end up. I think that the responses should bear structure: if registering REST handlers - list of handlers, if registering custom setting - list of settings, ...

Reply:
I'll remove ExtensionStringResponse and replace occurrences with AcknowledgedResponse